### PR TITLE
sach91 bootcamp week7 exercise

### PR DESCRIPTION
**Pull request by sach91**
Sachin's pull request for week 7 exercise in bootcamp.

**Finetuned Llama3.1-8b model on a small toys and games dataset**

1. Using the course notebooks, first created a lite dataset for toys and games from the Amazon Reviews dataset, having prices $1 - $60, uniform-sampling from each $1 bucket, having 1000 and 200 test samples. The prices are hence uniformly spread throughout the space.
_Dataset_: https://huggingface.co/datasets/sach91/lite-data
_Colab_: https://colab.research.google.com/drive/1f8eLrWYBCIhrmKTJPeKLKk4pqwfgzIJ-

2. Tested the traditional ML models from week6-exercise. As data is uniformly spread, most approaches' MAE is similar to the average of training samples.
- Random :	20.2
- Average Price:	15.8
- LR :		15.9
- LR + Vector :	14.1
- LR + W2V :	15.3
- XGB : 		13.4
- XGB + W2V :	14.8
- NER : 		16.5
- SVM : 		15.4
- RF :		14.8
- Expert (Avg 2): 13.3
- Expert (Avg 3): 13.1
- Scaling: no impact
\
Python Notebook: Committed to repo
_TrainData_: https://drive.google.com/file/d/180ZI9OIdivkO0T-H1wki1-K514iYYK_n
_TestData_: https://drive.google.com/file/d/1cW5doBO4jpbLQfZwygKFhy0CSSf6U1MQ

3. Tested the base model llama3.1-8b on this dataset.
MAE is 34.8, even higher then traditional models.
_Colab_: https://colab.research.google.com/drive/1mCJbF5x5zsA9dvr2dZZYvqdNk5h3AyCo

4. Fined tuned the base model using batch size of 4, in 250 steps (1 epoch) get below reduction in train loss.
_Model_: https://huggingface.co/sach91/pricer-2025-10-29_19.19.31
_Colab_: https://colab.research.google.com/drive/1wPvVYqEd6DeNqJUJ1i9Hwu-jujjqW4n8
image

5. Tested the finetuned model.
MAE reduced to 10.6 -> Just 1 epoch finetuning worked !
_Colab_: https://colab.research.google.com/drive/1uZgT2ceECYr3GKWpZBNr8NFedSlu9ps8
image

**Insights**

1. Though having 8B parameters, base model was not good performing on the lite dataset.
2. It seems due to the prices being uniformly spread in the range.
3. Fine-tuning is so effective that just 1 epoch raised the model from worst to best performer.

### CODE REVIEW

**Comment by ed-donner** [comment]
This is absolutely brilliant. You kept things nice and simple - made a neat test set, did a bunch of fun tests, tried out XGB, tried some ensembles, and then crushed it with some fine tuning. Love this.
